python/ray/_private/utils.py

```diff
@@ -1940,16 +1940,43 @@ def parse_node_labels_json(
     return labels
 
 
+def is_valid_label_string(s):
+    if len(s) == 0:
+        return True
+    pattern = r"^[a-zA-Z0-9\-_./]+$"
```

Comment:
Actually after some thoughts. I feel having a limitation on the length makes sense. But for valid characters, it seems a bit arbitrary. Why don't we allow `:` or `+`? We are also not following k8s since k8s doesn't allow `\`. Also I don't think we should strictly follow k8s.

Should we just have length limitations?